### ISSUE TITLE
`no-unused-vars` reports global vars on wrong line

### ISSUE THREAD

**Issue by IanVS** [eslint/eslint#3617]
Consider:

```js
/*eslint no-unused-vars: 2*/
/*global some_used_var */

var x = 10;
```

This will give the result:

```
  4:1  error  "some_used_var" is defined but never used  no-unused-vars
  4:5  error  "x" is defined but never used              no-unused-vars
```

It's a bit of a quibble, but might it be possible to report the error for some_used_var on line 2 (where it is defined as a global) instead?

**Comment by ilyavolodin** [comment]
This would be tough, as globals can also be declared in .eslintrc as well, and we don't actually know where the global is coming from (comments are parsed before rules are executed).

**Comment by IanVS** [comment]
Can it just always be at the top of the file then?  `1:1` would be easier to immediately grok, I think.

**Comment by nzakas** [comment]
Globals in config files are omitted from `no-unused-vars`. We should be able to add a location to inline globals.

**Comment by mysticatea** [comment]
I will try to fix this.

**Comment by mysticatea** [comment]
I found a similar issue around `/*exported`.

```js
/* eslint no-undef: 2, no-unused-vars: 2 */
/* exported foo, bar */
var bar = 0;
```

In this case, should `foo` be reported by `no-undef`?

**Comment by IanVS** [comment]
:clap:
